### PR TITLE
Gsoc2020 Add ranges in Polyhedron_3

### PR DESCRIPTION
**Pull request by Sameeer12**
## Summary of Changes
Adds ranges in `Polyhedron` so that users can use C++11 range-based loops. This PR adds methods with vertex/halfedge/facet handle ranges and a test file `test_range_based_loops.cpp` to test new methods.

## Release Management
* Affected package(s): Polyhedron
* Feature/Small Feature (if any): enhancement

### CODE REVIEW

File: Polyhedron/doc/Polyhedron/CGAL/Polyhedron_3.h [hunk 1]
```diff
@@ -914,6 +914,24 @@ class Polyhedron_3 {
   */
   typedef unspecified_type Plane_iterator;
 
+  /*!
+    range type for iterating over all vertices, with a nested
+    type `iterator` that has as value type `Vertex_handle`
```
Comment (lrineau) [inline Polyhedron/doc/Polyhedron/CGAL/Polyhedron_3.h:919]:
```suggestion
    type `iterator` that has a value type `Vertex_handle`
```

File: Polyhedron/doc/Polyhedron/CGAL/Polyhedron_3.h [hunk 2]
```diff
@@ -1096,6 +1114,13 @@ class Polyhedron_3 {
   */
   Vertex_iterator vertices_end();
 
+  /*!
+  returns a range of iterators over vertices.
+  \note While the value type of `Vertex_iterator` is `Vertex`, the value type of
+        `All_vertex_handles::iterator` is `Vertex_handle`
+  */
+  Vertex_handles vertex_handles();
```
Comment (lrineau) [inline Polyhedron/doc/Polyhedron/CGAL/Polyhedron_3.h:1122]: The documentation says `All_vertex_handles` but the return type is `Vertex_handles`.

File: Polyhedron/doc/Polyhedron/CGAL/Polyhedron_3.h [hunk 3]
```diff
@@ -1118,6 +1150,13 @@ class Polyhedron_3 {
   */
   Facet_iterator facets_end();
 
+ /*!
+  returns a range of iterators over facets.
+  \note While the value type of `Facet_iterator` is `Vertex`, the value type of
+        `All_facet_handles::iterator` is `Facet_handle`
+  */
+  Facet_handles facet_handles();
```
Comment (lrineau) [inline Polyhedron/doc/Polyhedron/CGAL/Polyhedron_3.h:1158]: The documentation says `All_facet_handles` but the return type is `Facet_handles`.

**Comment by lrineau** [comment]
That looks good. There should be a small feature for it. And I do not think that can goes as a bug-fix in CGAL-5.0.x, but only in `master`.

**Comment by danston** [comment]
Substituted by the PR #4845.

**Pull request closed without merging**